pkg/yaml/yq.go

```diff
@@ -110,3 +111,25 @@ func (e *Editor) SetValue(path string, value string) error {
 
 	return nil
 }
+
+// WalkNodes executes f for all yaml nodes found in path.
+// If an error is returned from f, the WalkNodes function will return the error and stop interating through
+// the rest of the nodes.
+func (e *Editor) WalkNodes(ctx context.Context, path string, f func(ctx context.Context, nc *yqlib.NodeContext) error) error {
```

Comment:
From my perspective, I don't see a ton of value to write a unit test for this function. We don't have much porter logic here outside of looping through the returned value and calling a callback function on the return value. What type of test cases do you have in mind for it?

Reply:
My general perspective is that over time other high level tests (you mentioned archive_test.go) may change and no longer give you coverage on a particular function, or be checking all the relevant things that a unit test would. Maybe it works out and is checking everything we care about, but I prefer to not rely on that as a test strategy for a codebase.

Sometimes we end up having to do that because the functionality in question is difficult to isolate, or involves a lot of coordination between other systems, requires a lot of setup, or is the type of functionality that would break so violently that it's obvious when it's not working. Those don't get unit tests and we rely on other types of testing.

But for something that is fairly isolated and straightforward to test, I prefer to see unit tests for those. Doesn't have to be exhaustive. You don't see me testing most error handling for instance unless it's important, for example the error message is being displayed to the user and we want to make sure it provides critical instructions. But I would like to see _some_ tests.

For the function below, I would at least be looking for a test that made sure I can open an editor, load a file, walk the nodes to modify something, and make sure when the file is written out that the yaml was modified as expected.

Reply:
Thank you for the explanation! I can write a test that make sure that the nodes are indeed traversed. However, I think the modification of a node is from the callback function, which the `generateManifest_test.go` should already covered that part